src/test/java/it/gov/pagopa/reward/service/reward/InitiativesEvaluatorServiceImplTest.java

```diff
@@ -0,0 +1,86 @@
+package it.gov.pagopa.reward.service.reward;
+
+import it.gov.pagopa.reward.dto.Reward;
+import it.gov.pagopa.reward.dto.RewardTransactionDTO;
+import it.gov.pagopa.reward.dto.TransactionDTO;
+import it.gov.pagopa.reward.model.counters.InitiativeCounters;
+import it.gov.pagopa.reward.model.counters.UserInitiativeCounters;
+import it.gov.pagopa.reward.test.fakers.TransactionDTOFaker;
+import it.gov.pagopa.reward.test.utils.TestUtils;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.math.BigDecimal;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class InitiativesEvaluatorServiceImplTest {
+
+    @Test
+    void evaluateInitiativesBudgetAndRules() {
+
+        // Given
+        RuleEngineService ruleEngineService = Mockito.mock(RuleEngineServiceImpl.class);
+
+        TransactionDTO trx = TransactionDTOFaker.mockInstance(1);
+        List<String> initiatives = List.of("INITIATIVE1");
+
+        InitiativeCounters initiativeCounters = InitiativeCounters.builder().initiativeId("INITIATIVE1").build();
+        UserInitiativeCounters userCounters = UserInitiativeCounters.builder()
+                .userId("USER1")
+                .initiatives(Map.of("INITIATIVE1", initiativeCounters))
+                .build();
+
+        Reward reward = new Reward(new BigDecimal("200"));
+        RewardTransactionDTO rTrx = RewardTransactionDTO.builder()
+                        .rewards(Map.of("INITIATIVE1", reward)).build();
+        Mockito.when(ruleEngineService.applyRules(Mockito.same(trx), Mockito.eq(initiatives), Mockito.eq(userCounters))).thenReturn(rTrx);
+
+        InitiativesEvaluatorService initiativesEvaluatorService = new InitiativesEvaluatorServiceImpl(ruleEngineService);
+
+        // When
+        RewardTransactionDTO result = initiativesEvaluatorService.evaluateInitiativesBudgetAndRules(trx, initiatives, userCounters);
+
+        // Then
+        Assertions.assertNotNull(result);
```

Comment:
just this check is not enough!

as you can see, it's the same as the below test, but you are not checking if the rejection reason is filled or not this time!